File: SWE4302_OOP_2/Lab_03/VehicleMS/src/Bus.java
public class Bus extends Vehicle implements IDriveable{
    private final FuelType fuelType;

    public Bus(String name, FuelType fuelType) {
        super(name);
        this.fuelType = fuelType;
    }

    public FuelType getFuelType() {
        return fuelType;
    }
    @Override
    public void AccelerateEngine() {
        System.out.println(name + " bus is accelerating engine with " + getFuelType() + " fuel");
    }
}

File: SWE4302_OOP_2/Lab_03/C_Sharp_VehicleMS/VehicleMS/Bus.cs
namespace VehicleMS
{
    public enum FuelType
    {
        DIESEL,
        PETROL,
        GASOLINE
    }

    public class Bus : Vehicle, IDriveable
    {
        private readonly FuelType fuelType;

        public Bus(string name, FuelType fuelType) : base(name)
        {
            this.fuelType = fuelType;
        }

        public void AccelerateEngine()
        {
            Console.WriteLine(Name + " bus is accelerating engine with " + fuelType + " fuel");
        }
    }
}
